lib/chart/api_velocity_chart.es6.js

```diff
@@ -0,0 +1,159 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+let d3 = require('d3');
+
+function apiVelocityChartController($scope, $timeout) {
+  let createNewDiffView = function(minuend, subtrahend) {
+    this.newDiffView({minuend, subtrahend});
+  }.bind(this);
+
+  this.$postLink = function() {
+    // AngularJS does not seem to have reliable post-render callbacks.
+    // Use $timeout as a hack.
+    $timeout(function() {
+      let width = 720;
+      let height = 360;
+      let margin = {top: 20, right: 50, bottom: 30, left: 50};
+      // Create a sub div inside charts div.
+      let div = d3.select(`#${this.chartId}`).append('div');
+      // Create an svg element inside div.
+      let svg = div.append('svg')
+          .attr('width', width)
+          .attr('height', height);
+      // Append title to SVG.
+      svg.append('text')
+          .attr('x', (width / 2))
+          .attr('y', margin.top)
+          .attr('text-anchor', 'middle')
+          .style('font-size', '20px')
+          .style('text-decoration', 'underline')
+          .text(this.title);
+      // Set actual chart width and height.
+      width = svg.attr('width') - margin.left - margin.right;
+      height = svg.attr('height') - margin.top - margin.bottom;
+      // Create a g element inside svg,
+      // which will contain all paths and areas.
+      let g = svg.append('g').attr('transform',
+          'translate(' + margin.left + ',' + margin.top + ')');
+      let rightAxisUpperBound = Math.ceil(d3.max(
+          this.apiVelocity, (d) => d.totalApis + d.removedApis) / 500) * 500;
+      let rightAxisLowerBound = Math.floor(d3.min(
+          this.apiVelocity, (d) => d.totalApis - d.newApis) / 500) * 500;
+      // X axis is release date.
+      let x = d3.scaleTime()
+          .rangeRound([0, width])
+          .domain(d3.extent(this.apiVelocity, (d) => d.releaseDate));
+      // Y axis is number of APIs.
+      let y = d3.scaleLinear()
+          .rangeRound([height, 0])
+          .domain([rightAxisLowerBound, rightAxisUpperBound]);
+      // Append tooltip div inside chart div. (cannot append div inside svg)
+      let tooltips = d3.select(`#${this.chartId}-tooltip`);
+      // Total number of APIs' area.
+      let areaTotal = d3.area()
+          .x((d) => x(d.releaseDate))
+          .y1((d) => y(d.totalApis))
+          .y0(y(rightAxisLowerBound));
+      // Number of removed APIs.
+      let areaRemovedAPI = d3.area()
+          .x((d) => x(d.releaseDate))
+          .y1((d) => y(d.totalApis + d.removedApis))
+          .y0((d) => y(d.totalApis));
+      // Number of new APIs.
+      let areaNewAPI = d3.area()
+          .x((d) => x(d.releaseDate))
+          .y1((d) => y(d.totalApis))
+          .y0((d) => y(d.totalApis - d.newApis));
+      // Create dot where tooltips are displayed.
+      svg.selectAll('dot')
+          .data(this.apiVelocity)
+        .enter().append('circle')
```

Comment:
This is d3 style.  append('circle') is +2 indent because the following attr methods are describing the circle, while the previous data method are used on 'dot'.